src/ray/core_worker/core_worker.cc

```diff
@@ -1415,9 +1415,16 @@ Status CoreWorker::ExperimentalChannelWriteAcquire(
     const std::shared_ptr<Buffer> &metadata,
     uint64_t data_size,
     int64_t num_readers,
-    std::shared_ptr<Buffer> *data) {
-  return experimental_mutable_object_provider_->WriteAcquire(
-      object_id, data_size, metadata->Data(), metadata->Size(), num_readers, *data);
+    std::shared_ptr<Buffer> *data,
+    int64_t timeout_ms) {
+  int64_t expiration_ms = timeout_ms == -1 ? -1 : current_time_ms() + timeout_ms;
```

Comment:
Could we move this into `WriteAcquire` and have `WriteAcquire` take a timeout? That way we can have a special case for 0 timeout (return if result is already ready else throw error).